### ISSUE TITLE
daedalus wallet

### ISSUE THREAD

**Issue by prodigydefender** [input-output-hk/cardano-sl#2128]
downloaded the wallet, watched the video tutorial at homepage, my downloaded wallet is taking forever to sync blocks? this isn't shown in the tutorial at homepage. anyone knows?

**Comment by bjwebber88** [comment]
I am having same issue

**Comment by bjwebber88** [comment]
its at 99.97... Strange

**Comment by fschr** [comment]
It's seriously ridiculous that users must wait until blocks are sync'd before using the wallet.

What if I want to quickly generate a private key? What if I want to look at addresses associated to my wallet? What if I want to remove addresses? All of these extremely necessary functionalities **do not require a network connection**, and yet, Daedalus is designed so that one must wait for blocks to sync before making use of them.

IOHK team: providing a means to safely generate private keys and addresses should be a number one priority. I shouldn't have to "sync blocks," build a web app, or search through a mountain of documentation to figure out how to safely generate a private key on my computer.

**Comment by wiatrM** [comment]
I agree, when you are planning to announce light wallet or fast wallet (keys) generator?

**Comment by fschr** [comment]
@wiatrM based on another issue (#2140), I believe a CLI wallet and a CLI key generator already ship with cardano-SL. The documentation is just out of date and it doesn't say how to build them. I'll let you know when I figure it out.
